### ISSUE TITLE
Exposing the output of `az --version` in JSON and other parsable formats.

### ISSUE THREAD

**Issue by nascarsayan** [Azure/azure-cli#27457]
**Related command**

`az --version`

**Is your feature request related to a problem? Please describe.**

`az --version` outputs several valuable information apart from the current versions of installed `azure-cli` modules and extensions. This includes:
1. Which code CLI modules are not up-to-date (`azure-cli` , `azure-cli-core`, `azure-cli-telemetry`).
2. Path to the underlying `python` used by `az`. The mode of installation of this python.
3. Path to the extensions directory.

All this information is not available in a code-parsable format. We currently need to do string parsing which can be flaky.

**Describe the solution you'd like**

`az --version -o json --query '<>` can be used to query a specific property of the `az` installation. We can decide on the schema of the exposed property.

**Describe alternatives you've considered**

Currently, I need to parse the string output, using regex, etc. to get the python installation location, and also to check if upgrade is available.

**Additional context**

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by jiasli** [comment]
We already have `az version` for similar purpose (https://github.com/Azure/azure-cli/pull/11680, https://github.com/Azure/azure-cli/pull/13001).

```
> az version
{
  "azure-cli": "2.51.0",
  "azure-cli-core": "2.51.0",
  "azure-cli-telemetry": "1.1.0",
  "extensions": {
    "next": "0.1.3",
    "ssh": "1.1.6",
    "storage-preview": "0.8.3"
  }
}
```

To better help us understand your needs, could you let us know:

- Which properties do you want `az version` to include?
- Do you have a proposal or recommendation about how they should look like?

**Comment by nascarsayan** [comment]
Yes, I'm aware of `az version` command.
I'm fine with either of the following:
- adding the support for formatted output (like json) to `az --version`
- adding the support of displaying all the extra info through `az version` itself.

In case it's easier to change the `az version` command, something like this could be the output:

`$ az version --full`
```json
{
  "azure-cli": "2.51.0",
  "azure-cli-core": "2.51.0",
  "azure-cli-telemetry": "1.1.0",
  "extensions": {
    "next": "0.1.3",
    "ssh": "1.1.6",
    "storage-preview": "0.8.3"
  },
  "source": {
    "python": {
      "location": "C:\\Program Files\\Microsoft SDKs\\Azure\\CLI2\\python.exe",
      "version": "Python (Windows) 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)]"
    },
    "extensions": {
      "location": "C:\\Users\\contosouser\\.azure\\cliextensions",
      "updates": {
        "azure-cli": "2.52.0",
        "azure-cli-core": "2.52.0",
        "azure-cli-telemetry": "1.1.0"
      }
    }
  }
}
```

**Comment by jiasli** [comment]
Thank you @nascarsayan for the information. Could you also share the use case of these properties? Then we will evaluate with our PMs whether we want to add them.

**Comment by nascarsayan** [comment]
Sure!
- python location : We need to evaluate if the python is 32-bit or 64-bit MSI. `az arcappliance vmware` command requires 64-bit python, so our onboarding script has to extract the python location of the azure-cli, and run:
  ```powershell
  $azVersion = az --version *>&1;
  $azVersionLines = $azVersion -split "`n"
  $pyLoc = $azVersionLines | Where-Object { $_ -match "^Python location" }
  $pythonExe = $pyLoc -replace "^Python location '(.+?)'$", '$1'
  $arch = & $pythonExe -c "import struct; print(struct.calcsize('P') * 8)";
  # assert($arch -ge 64);
  ```
  If the python is not 64-bit, we need to download the 64-bit MSI, and install it.
- updates: Since `az upgrade` does not currently provide a non-interactive way of upgrading the azure CLI installed as MSI, we need to detect if updates for `azure-cli` core packages are available. If upgrade is available, we download the latest MSI and install it on top of the current `az`.

For our usecase, we don't require the folder location of extensions, and the python version (since 64-bit is hard to parse from the text output). However, we could add these details for completeness. I am flexible on the new schema, as long as the information can be extracted.

**Comment by jiasli** [comment]
@nascarsayan, while it is true you can detect if the `python.exe` is 32-bit or 64-bit with [`struct.calcsize`](https://docs.python.org/3/library/struct.html#struct.calcsize):

```
> & "C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe" -c "import struct; print(struct.calcsize('P') * 8)"
64
```

(I guess you find this approach from https://stackoverflow.com/questions/1405913/how-do-i-determine-if-my-python-shell-is-executing-in-32bit-or-64bit), it actually doesn't need to be that complicated.

The 64-bit MSI is installed under `C:\Program Files` (`ProgramFiles64Folder`):

https://github.com/Azure/azure-cli/blob/aeb94ed12a46acdd14e7c4e2633908e149f4a1bc/build_scripts/windows/Product.wxs#L16

and 32-bit MSI is installed under `C:\Program Files (x86)` (`ProgramFilesFolder`):

https://github.com/Azure/azure-cli/blob/aeb94ed12a46acdd14e7c4e2633908e149f4a1bc/build_scripts/windows/Product.wxs#L26

So you may simply get the installation location of Azure CLI with:

```
> (Get-Command az).Source
C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin\az.cmd
```

then detect 64/32-bit based on the installation path.
